### PR TITLE
Cleaning up SaaS tests

### PR DESCRIPTION
**Pull request by galvana**
# Purpose

Cleaning up SaaS tests to use a sample SaaS config for unit tests.

# Changes
- Added example_saas_config.yml and example_saas_dataset.yml
- Created new pytest marks `integration_mailchimp` and `unit_saas`
- Needs to replace the `MAILCHIMP_ACCOUNT_EMAIL` with `MAILCHIMP_IDENTITY_EMAIL` in the Github secrets to better reflect the use case for the email

# Checklist

- [ ] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md))
- [ ] Good unit test/integration test coverage
- [ ] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

### CODE REVIEW

File: tests/api/v1/endpoints/test_connection_config_endpoints.py [hunk 1]
```diff
@@ -814,18 +814,18 @@ def test_put_http_connection_config_secrets(
         assert https_connection_config.last_test_timestamp is None
         assert https_connection_config.last_test_succeeded is None
 
-    @pytest.mark.saas_connector
-    def test_put_connection_config_saas_secrets(
+    @pytest.mark.unit_saas
```
Comment (seanpreston) [inline tests/api/v1/endpoints/test_connection_config_endpoints.py:817]: What's the purpose of this marker? It looks like we aren't invoking it anywhere, is it a util you've been using for local testing?

Comment (galvana) [reply tests/api/v1/endpoints/test_connection_config_endpoints.py:817]: That's right, it's something I've been using to quickly target the SaaS tests during development.

File: tests/fixtures/saas_fixtures.py [hunk 2]
```diff
@@ -40,38 +45,63 @@ def load_config(filename: str) -> Dict:
 
 
 @pytest.fixture
-def example_saas_configs() -> Dict[str, Dict]:
-    example_saas_configs = {}
-    example_saas_configs["mailchimp"] = load_config(
-        "data/saas/config/mailchimp_config.yml"
+def saas_configs() -> Dict[str, Dict]:
```
Comment (seanpreston) [inline tests/fixtures/saas_fixtures.py:48]: This was actually a pattern I was hoping we could move away from. It makes more sense to address each config in a fixture individually I think, so we don't need to load in every single one and address it opaquely as a list index.

File: tests/fixtures/saas_fixtures.py [hunk 3]
```diff
@@ -40,38 +45,63 @@ def load_config(filename: str) -> Dict:
...
+def saas_datasets() -> Dict[str, Dict]:
+    saas_datasets = {}
+    saas_datasets["saas_example"] = load_dataset(
+        "data/saas/dataset/saas_example_dataset.yml"
+    )[0]
+    saas_datasets["mailchimp"] = load_dataset(
         "data/saas/dataset/mailchimp_dataset.yml"
     )[0]
-    return example_saas_datasets
+    return saas_datasets
```
Comment (seanpreston) [inline tests/fixtures/saas_fixtures.py:66]: As above

File: tests/fixtures/saas_fixtures.py [hunk 4]
```diff
@@ -142,27 +194,29 @@ def connection_config_saas_with_invalid_saas_config(
...
+    return pydash.get(saas_config, "mailchimp.identity_email") or os.environ.get(
+        "MAILCHIMP_IDENTITY_EMAIL"
     )
 
 
 @pytest.fixture(scope="function")
-def reset_mailchimp_data(connection_config_saas, mailchimp_account_email) -> Generator:
+def reset_mailchimp_data(
+    connection_config_mailchimp, mailchimp_identity_email
+) -> Generator:
```
Comment (seanpreston) [inline tests/fixtures/saas_fixtures.py:210]: We don't necessarily have to do this here, but ultimately we're going to want to keep these fixtures in their own files akin to datastores, e.g. `../mailchimp_fixtures.py`

**Review by seanpreston — APPROVED** [review]

**Pull request merged**